fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/BoundFilter.java

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.druid.model.filter;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import javax.validation.constraints.Null;
+import java.util.Objects;
+
+/**
+ * Bound filters supports filtering on ranges of dimension values.
+ */
+public class BoundFilter extends DimensionalFilter<BoundFilter> {
+
+    /**
+     * Ordering specified for the range filtering.
+     */
+    public enum Ordering {
+        LEXICOGRAPHIC,
+        ALPHANUMERIC,
+        NUMERIC,
+        STRLEN;
+
+        final String orderingType;
+
+        /**
+         * Constructor.
+         */
+        Ordering() {
+            this.orderingType = EnumUtils.enumJsonName(this);
+        }
+
+        /**
+         * Get the JSON representation of this class.
+         *
+         * @return the JSON representation.
+         */
+        @JsonValue
+        public String toJson() {
+            return orderingType;
+        }
+    }
+
+    private final String lower;
+    private final String upper;
+    private final Boolean lowerStrict;
+    private final Boolean upperStrict;
+    private final Ordering ordering;
+
+    /**
+     * Constructor.
+     *
+     * @param dimension The druid dimension to be filtered
+     * @param lower The lower bound of the dimension value to be filtered (Optional)
+     * @param upper The upper bound of the dimension value to be filtered (Optional)
+     * @param lowerStrict Boolean to enable/ disable strict filtering for lower bounds (Optional)
+     * @param upperStrict Boolean to enable/ disable strict filtering for upper bounds (Optional)
+     * @param ordering The Ordering to be applied for the dimension filtering
+     */
+    public BoundFilter (
+            Dimension dimension,
+            @Null String lower,
+            @Null String upper,
+            @Null Boolean lowerStrict,
+            @Null Boolean upperStrict,
+            @Null Ordering ordering
+    ) {
+        super(dimension, DefaultFilterType.BOUND);
+        this.lower = lower;
+        this.upper = upper;
+        this.lowerStrict = lowerStrict;
+        this.upperStrict = upperStrict;
+        this.ordering = ordering;
+    }
+
+    public String getLower() {
+        return this.lower;
+    }
+
+    public String getUpper() {
+        return this.upper;
+    }
+
+    public Boolean isLowerStrict() {
+        return this.lowerStrict;
+    }
+
+    public Boolean isUpperStrict() {
+        return this.upperStrict;
+    }
+
+    public Ordering getOrdering() {
```

Comment:
hmm...No Javadoc should trigger checkstyle to fail... @michael-mclawhorn We are doing something wrong in our checkstyle config

Reply:
http://checkstyle.sourceforge.net/config_javadoc.html#JavadocMethod

Currently for getter methods, we do not need to provide java docs because of this property `allowMissingPropertyJavadoc`.

https://github.com/yahoo/fili/blob/master/checkstyle-style.xml#L186

Reply:
Yeah, I missed this too, but we have a lot of simple getters without javadoc in our codebase.